### PR TITLE
updated gen2 relationship docs with new relationship behavior

### PR DESCRIPTION
**Pull request by renebrandel**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 1]
```diff
@@ -104,110 +114,164 @@ const schema = a.schema({
     tags: a.manyToMany('Tag', { relationName: 'PostTags'})
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:114]: Not part of this PR, but the Update "Has Many" example on L70 should use a different ID for the newly updated member. Perhaps the "Create" example can use `"ORIGINAL_MEMBER_ID"`, the "Update" example can use `"OTHER_MEMBER_ID"`, and the "Delete" example can use `"ORIGINAL_MEMBER_ID"`. That will reinforce the expectation that customers are responsible for maintaining the relationship integrity.

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 2]
```diff
@@ -104,110 +114,164 @@ const schema = a.schema({
...
+### Eager load a "many-to-many" relationship
+
+```ts
+const { data: post } = await client.models.Post.get({
+  id: window.prompt("ID") ?? ""
+}, {
+  selectionSet: ["tags.*", "tags.tag.*"]
+})
+
+const tags = post.tags.map(postTag => postTag.tag)
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:190]: nit, for consistency with above example

```suggestion
const { data: postWithTags } = await client.models.Post.get({
  id: window.prompt("ID") ?? ""
}, {
  selectionSet: ["tags.*", "tags.tag.*"]
})

const tags = postWithTags.tags.map(postTag => postTag.tag)
```

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 3]
```diff
@@ -104,110 +114,164 @@ const schema = a.schema({
...
 ### Create a "Has One" relationship between records
 
-To create a "has one" relationship between records, first create the parent item and then create the child item and assign the parent. 
+To create a "has one" relationship between records, first create record on the target model and then create a record on the origin model and assign the target model's record. 
 
 ```ts
-const { data: team } = await client.models.Team.create({
-  name: 'My Team',
+const { data: cart } = await client.models.Cart.create({
+  items: ["Eggs", "Cilantro"]
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:222]: I don't know what kind of weird-ass quiche you're making, but count me out.

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 4]
```diff
@@ -104,110 +114,164 @@ const schema = a.schema({
...
+const { data: cart } = await customer.activeCart();
 ```
 
 ### Eagerly load a "Has One" relationship
 
 ```ts
-const { data: project } = await client.models.Project.get(
-  { id: "MY_PROJECT_ID" },
-  { selectionSet: ['id', 'team.*'] },
+const { data: customer } = await client.models.Customer.get(
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:269]: as above:

```suggestion
const { data: customerWithCart } = await client.models.Customer.get(
```

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 5]
```diff
@@ -104,110 +114,164 @@ const schema = a.schema({
...
+  { id: "MY_CUSTOMER_ID" },
+  { selectionSet: ['id', 'activeCart.*'] },
 );
 ```
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:273]: Do you want to fill out the example with a usage of the eagerly loaded item?

````suggestion
const cart = customerWithCart.activeCart;
```
````

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 6]
```diff
@@ -220,78 +284,109 @@ You can make a "has one" or "has many" relationship bidirectional with the `belo
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:284]: such picky, very nit: Since you introduce `hasMany` then `hasOne` above, maybe reverse the order of the bidirectional examples to match?

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 7]
```diff
@@ -220,78 +284,109 @@ You can make a "has one" or "has many" relationship bidirectional with the `belo
...
+</Block>
+
+</BlockSwitcher>
+
 ```ts
 const client = generateClient<Schema>();
 
-const { data: team } = await client.models.Team.get({ id: 'FABF27EE-25D2-4851-AE1C-09BCA1604C77' });
+const { data: customer } = await client.models.Customer.get({
+  id: 'FABF27EE-25D2-4851-AE1C-09BCA1604C77'
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:314]: nit
```suggestion
  id: 'SOME_CUSTOMER_ID'
```

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 8]
```diff
@@ -220,78 +284,109 @@ You can make a "has one" or "has many" relationship bidirectional with the `belo
...
+});
 
-console.log((await team.project()).data?.name); // There is a 'project' field in the team
+const { data: cart } = await customer.activeCart();
 
-const { data: teamProject } = await team.project();
-const { data: teamProjectRelatedTeam } = await teamProject.team();
-// teamProjectRelatedTeam is equivalent to original team
+// references back to the same customer as above
+const { data: belongsToCustomer } = await cart.customer();
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:320]: It should be clear from context, but do you want an explicit eager/lazy loading example in the bidi examples?

File: src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx [hunk 9]
```diff
@@ -104,110 +114,164 @@ const schema = a.schema({
...
 ```ts
-const { data: newTeam } = await client.models.Team.create({
-  name: 'New Team',
+const { data: newCart } = await client.models.Cart.create({
+  items: ["Mint"],
 });
 
-await client.models.Project.update({
+await client.models.Customer.update({
   id: project.id,
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/data-modeling/relationships/index.mdx:243]:
```suggestion
  id: customer.id,
```

File: public/images/gen2/data-modeling/has-many-member-to-team.png
[the remark this answers is not in the document]
Comment (renebrandel) [reply public/images/gen2/data-modeling/has-many-member-to-team.png:1]: Good suggestions will do!

**Comment by renebrandel** [comment]
closing in favor of https://github.com/aws-amplify/docs/pull/7222
